Lib/test/test_ctypes/test_bitfields.py

```diff
@@ -186,8 +186,10 @@ class X(Structure):
                 self.assertEqual((c_typ, x.a, x.b, x.c), (c_typ, 0, 7, 0))
 
     def fail_fields(self, *fields):
-        return self.get_except(type(Structure), "X", (),
-                               {"_fields_": fields})
+        for layout in "ms", "gcc-sysv":
+            with self.subTest(layout=layout):
+                return self.get_except(type(Structure), "X", (),
```

Comment:
Nnnno, good point!
Bitfields in unions are rather pointless, but they should be tested anyway.

Reply:
Ahahaha, turns out MSC and GCC with `__attribute__((ms_struct))` disagree on the alignment of unions with bitfields.
Guess that goes to show that it's not an important use case. I'll test with single-byte types for now to avoid complicating this PR.

Reply:
I'd prefer to look at #117942 separately, and to add a dedicated test for it.

Reply:
> Ahahaha

Sorry for ruining your mood with my evil questions :smiling_imp:. I think we need to add some comment somewhere just to remember that MSC and GCC don't agree for that (otherwise I'll definitely forget about it).

Reply:
Hm, guess I'll make a new issue for bitfields in unions, and handle it separately.

Tomorrow. I'm out for today. Thanks for the reviews so far!

Reply:
You're welcome! and thank you for keeping up with my nitpicky reviews (I'm very nitpicky and I know it can annoy people, so please tell me if this is the case)

Reply:
Let's use #83588.

Your reviews are fine! I don't really care about sometimes missing the exact right number of spaces in indentation, but if you send a suggestion it's easy enough to apply :)